src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,56 @@
+
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    public static final DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
+            throws DataTimeException, IllegalArgumentException {
+        StringBuilder builder = new StringBuilder();
+        for (int i = 0; i < names.length; i++) {
+            String name = "";
+            int resultSum = 0;
+            for (int j = 0; j < data.length; j++) {
+                String[] split = data[j].split(" ");
+                try {
+                    LocalDate fromThisDate = LocalDate.parse(dateFrom, dtf);
+                    LocalDate toThisDate = LocalDate.parse(dateTo, dtf);
+                    LocalDate date = LocalDate.parse(split[0], dtf);
+                    if (date.isEqual(fromThisDate) || date.isEqual(toThisDate)
+                            || date.isAfter(fromThisDate) && date.isBefore(toThisDate)) {
+                        if (names[i].equals(split[1])) {
+                            int parseInt = Integer.parseInt(split[2]);
+                            int parseInt2 = Integer.parseInt(split[3]);
```

Comment:
parseInt, parseInt2 - I can not understand meaning of these names

2, 3 - **Any magic numbers should be constants**
[Common mistakes](https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info#any-magic-numbers-should-be-constants)